### DISCUSSION TITLE
How does @babel/preset-env transpile the JS code based on Browserslist?

### DISCUSSION THREAD

**Discussion by mrdulin** [babel/babel#16607] [Q&A]
For example, the `"chrome: 49"` query. My understand is babel will transpiled the JS code compatible with Chrome 49 version.

How does `@babel/preset-env` know what kind of transpiled JS code should be emited? What babel plugin, polyfill should be used?

The query is very fragmented, and it seems impossible to use a mapping of query and plugins, so babel can get the needed plugins by query.

For example：

- babel will include plugin-a, plugin-b, polyfill-a, polyfill-b for `"chrome 49"` query.
- babel will include plugin-a, polyfill-a for `"chrome 86"` query.

I didn't find the source code or any doc explain this process.

**Comment by nicolo-ribaudo** [comment]
We use data from this package: https://github.com/babel/babel/tree/main/packages/babel-compat-data
Which is based on https://github.com/kangax/compat-table/

**Reply by mrdulin** [reply]
Thanks. It's helpful. It seems babel does have a map of query and plugins when I see this file [plugins.json](https://github.com/babel/babel/blob/main/packages/babel-compat-data/data/plugins.json)
